webpush/encrypt.go

```diff
@@ -296,8 +298,11 @@ func encrypt(plaintext, key, nonce []byte) ([]byte, error) {
 
 // Given the coordinates of a party A's public key and the bytes of party B's
 // private key, compute a shared secret.
-func sharedSecret(curve elliptic.Curve, pub, priv []byte) []byte {
+func sharedSecret(curve elliptic.Curve, pub, priv []byte) ([]byte, error) {
 	publicX, publicY := elliptic.Unmarshal(curve, pub)
+	if publicX == nil {
+		return nil, fmt.Errorf("Invalid keys provided")
```

Comment:
I would provide a bit more info. "invalid keys" can be anything and quite unhelpful. you're expecting a certain component of the key to be present and can just say so in the error.